@geoand  

Commit 59b63bba0 adds a single new integration test that has in its application.properties:
```
quarkus.lambda.handler=test
quarkus.native.remote-container-build=true
```

It tests that both #41020 is fixed and that #35718 has no regression.

Tested as 
```
$ ./mvnw clean verify -f integration-tests -pl awt-packaging -Dnative -Dnative.surefire.skip=false
```

It passes, although **it produces these Lambda server related warnings**, I guess it could have something to do with the fact that the LambdaClient is deprecated...?

I tried to ask about it on Zulip:  https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Testing.20AWS.20Lambda.20locally/near/444730456 

```
[INFO] --- failsafe:3.2.5:integration-test (default) @ quarkus-integration-test-awt-packaging ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.jaxb.AwtJaxbTestIT
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
getting verticle
Executing "/home/karm/workspaceRH/quarkus/integration-tests/awt-packaging/target/quarkus-integration-test-awt-packaging-999-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/home/karm/workspaceRH/quarkus/integration-tests/awt-packaging/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus-internal.aws-lambda.test-api=localhost:5387"
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2024-06-14 18:42:08,760 INFO  [io.quarkus] (main) quarkus-integration-test-awt-packaging 999-SNAPSHOT native (powered by Quarkus 999-SNAPSHOT) started in 0.015s. Listening on: http://0.0.0.0:8081
2024-06-14 18:42:08,760 INFO  [io.quarkus] (main) Profile prod activated. 
2024-06-14 18:42:08,760 INFO  [io.quarkus] (main) Installed features: [amazon-lambda, awt, cdi, resteasy, resteasy-jaxb, smallrye-context-propagation, vertx]
2024-06-14 18:42:08,760 INFO  [io.qua.ama.lam.run.AbstractLambdaPollLoop] (Lambda Thread (NORMAL)) Listening on: http://localhost:5387/2018-06-01/runtime/invocation/next
2024-06-14 18:42:09,032 WARN  [io.qua.ama.lam.tes.LambdaClient] (main) LambdaClient has been deprecated and will be removed in future Quarkus versions.  You can now invoke using a built in test http server.  See docs for more details
2024-06-14 18:42:09,664 INFO  [io.qua.it.jax.Resource] (executor-thread-1) Received book: io.quarkus.it.jaxb.Book@2a3173ef
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.712 s -- in io.quarkus.it.jaxb.AwtJaxbTestIT
2024-06-14 18:42:12,661 ERROR [io.qua.ama.lam.run.AbstractLambdaPollLoop] (Lambda Thread (NORMAL)) Error running lambda (NORMAL): java.net.SocketException: Connection reset
        at java.base@21.0.3/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:318)
        at java.base@21.0.3/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
        at java.base@21.0.3/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
        at java.base@21.0.3/java.net.Socket$SocketInputStream.read(Socket.java:1099)
        at java.base@21.0.3/java.io.BufferedInputStream.fill(BufferedInputStream.java:291)
        at java.base@21.0.3/java.io.BufferedInputStream.read1(BufferedInputStream.java:347)
        at java.base@21.0.3/java.io.BufferedInputStream.implRead(BufferedInputStream.java:420)
        at java.base@21.0.3/java.io.BufferedInputStream.read(BufferedInputStream.java:399)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:827)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:759)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:952)
        at java.base@21.0.3/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:759)
        at java.base@21.0.3/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1690)
        at java.base@21.0.3/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1599)
        at java.base@21.0.3/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3235)
        at io.quarkus.amazon.lambda.runtime.AbstractLambdaPollLoop$1.run(AbstractLambdaPollLoop.java:95)
        at java.base@21.0.3/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21.0.3/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)

2024-06-14 18:42:12,661 ERROR [io.qua.run.StartupContext] (Lambda Thread (NORMAL)) Running a shutdown task failed [Error Occurred After Shutdown]: java.lang.RuntimeException: java.lang.InterruptedException
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$13.run(VertxHttpRecorder.java:888)
        at io.quarkus.runtime.StartupContext.runAllAndClear(StartupContext.java:87)
        at io.quarkus.runtime.StartupContext.close(StartupContext.java:79)
        at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
        at io.quarkus.runtime.Application.stop(Application.java:208)
        at io.quarkus.runtime.Application.stop(Application.java:155)
        at io.quarkus.amazon.lambda.runtime.AbstractLambdaPollLoop$1.run(AbstractLambdaPollLoop.java:165)
        at java.base@21.0.3/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21.0.3/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
Caused by: java.lang.InterruptedException
        at java.base@21.0.3/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1100)
        at java.base@21.0.3/java.util.concurrent.CountDownLatch.await(CountDownLatch.java:230)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$13.run(VertxHttpRecorder.java:886)
        ... 10 more

2024-06-14 18:42:12,662 INFO  [io.quarkus] (Lambda Thread (NORMAL)) quarkus-integration-test-awt-packaging stopped in 0.001s
2024-06-14 18:42:12,664 WARN  [io.net.boo.ServerBootstrap] (vert.x-acceptor-thread-0) Failed to register an accepted channel: [id: 0x94896071, L:/127.0.0.1:5387 ! R:/127.0.0.1:48170]: java.lang.IllegalStateException
        at io.vertx.core.net.impl.VertxEventLoopGroup.next(VertxEventLoopGroup.java:37)
        at io.vertx.core.net.impl.VertxEventLoopGroup.register(VertxEventLoopGroup.java:53)
        at io.netty.bootstrap.ServerBootstrap$ServerBootstrapAcceptor.channelRead(ServerBootstrap.java:241)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:97)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:553)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:840)

[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 

```

--- a/.github/native-tests.json
+++ b/.github/native-tests.json
@@ -145,9 +145,9 @@
             "os-name": "ubuntu-latest"
         },
         {
-            "category": "AWT, ImageIO and Java2D",
-            "timeout": 30,
-            "test-modules": "awt, no-awt",
+            "category": "AWT, ImageIO and Java2D, Packaging .so files",
+            "timeout": 40,
+            "test-modules": "awt, no-awt, awt-packaging",
             "os-name": "ubuntu-latest"
         }
     ]

